app/src/main/java/org/mozilla/fenix/collections/CollectionCreationFragment.kt

```diff
@@ -25,6 +25,9 @@ class CollectionCreationFragment : DialogFragment() {
     private lateinit var collectionCreationStore: CollectionCreationStore
     private lateinit var collectionCreationInteractor: CollectionCreationInteractor
 
+    private var _binding: FragmentCreateCollectionBinding? = null
+    private val binding get() = _binding!!
```

Comment:
should this be `!!`?

Reply:
I've personally only seen it as a lateinit var

Reply:
Fragments need special care since they have a more complex lifecycle.
In fragments that are manually detached or put to backstack (most of ours are) the View will be destroyed but the fragment instance lives on.
Since when coming back to the fragment (or when it is re-attached) onCreateView will be called again (and the binding should be initialized again) Android docs recommend the above idiom to:
- null the View Binding to reduce the memory pressure in between onDestroyView and onCreateView
- have an easy and safe way to access it, not having to check it for null when used while the Fragment is displayed (we control the lifecycle of `_binding` and know that while the View exists that null assertion is safe).

If the fragment is not put to backstack and does not outlive the View we should not need this idiom as when it will be destroyed it will cleanup all properties by itself.

There is an open issue to simplify the Fragment's lifecycle - https://issuetracker.google.com/issues/124181028.

Reply:
@eliserichards I also dislike using `!!` 😞

Reply:
This is also the recommended approach from https://developer.android.com/topic/libraries/view-binding#fragments